community/detectors/intel_neural_compressor_cve_2024_22476/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202422476/Cve202422476VulnDetector.java

```diff
@@ -0,0 +1,208 @@
+import java.time.Instant;
+import javax.inject.Inject;
+
+/** A {@link VulnDetector} that detects the CVE-2024-22476 vulnerability. */
+@PluginInfo(
+    type = PluginType.VULN_DETECTION,
+    name = "CVE-2024-22476 Detector",
+    version = "0.1",
+    description = "Checks for occurrences of CVE-2024-22476 in  Intel Neural Compressor instances.",
+    author = "frkngksl",
+    bootstrapModule = Cve202422476DetectorBootstrapModule.class)
+@ForWebService
+public final class Cve202422476VulnDetector implements VulnDetector {
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+
+  private final Clock utcClock;
+  private final HttpClient httpClient;
+  private final PayloadGenerator payloadGenerator;
+
+  private static final String VUL_PATH = "task/submit/";
+  private static final int BATCH_REQUEST_WAIT_AFTER_TIMEOUT = 10;
+  private final String taskRequestTemplate;
+
+  @Inject
+  Cve202422476VulnDetector(
+      @UtcClock Clock utcClock, HttpClient httpClient, PayloadGenerator payloadGenerator)
+      throws IOException {
+    this.utcClock = checkNotNull(utcClock);
+    this.httpClient =
+        checkNotNull(httpClient, "HttpClient cannot be null.")
+            .modify()
+            .setFollowRedirects(false)
+            .build();
+    this.payloadGenerator = checkNotNull(payloadGenerator, "PayloadGenerator cannot be null.");
+    taskRequestTemplate =
+        Resources.toString(Resources.getResource(this.getClass(), "task_request.json"), UTF_8);
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(Cve202422476VulnDetector::isWebServiceOrUnknownService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private static boolean isWebServiceOrUnknownService(NetworkService networkService) {
+    return networkService.getServiceName().isEmpty()
+        || NetworkServiceUtils.isWebService(networkService)
+        || NetworkServiceUtils.getServiceName(networkService).equals("unknown");
+  }
+
+  private static StringBuilder buildTarget(NetworkService networkService) {
+    StringBuilder targetUrlBuilder = new StringBuilder();
+    if (NetworkServiceUtils.isWebService(networkService)) {
+      targetUrlBuilder.append(NetworkServiceUtils.buildWebApplicationRootUrl(networkService));
+    } else {
+      targetUrlBuilder
+          .append("https://")
+          .append(toUriAuthority(networkService.getNetworkEndpoint()))
+          .append("/");
+    }
+    return targetUrlBuilder;
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    return isRceExecutable(networkService);
+  }
```

Comment:
This is redundant, you can implement the whole business logic into the `isServiceVulnerable` function.